src/store.js

```diff
@@ -139,24 +139,51 @@ export const useStore = defineStore("store", {
           this.isInputDisabled = true;
         });
     },
+    // toggleCard(breed, id) {
+    // breed.isActive = !breed.isActive;
+    // if (breed.isActive) {
+    //   this.indexClicked++;
+    //   this.lastBreedState.push(id);
+    // }
+    // this.displayedList.forEach((item) => {
+    //   if (item.name !== breed.name) {
+    //     item.isActive = false;
+    //   }
+    // });
+    // },
+
+    resetToDefault() {
+      this.lastBreedState = [];
+      this.indexClicked = 0;
+    },
+    resetSearchQuery() {
+      this.searchQuery = "";
+    },
     toggleCard(breed, id) {
+      this.setBreedIsActive(this.ifBreedActive(this.toggleIsActive(breed), id));
+    },
+
+    toggleIsActive(breed) {
       breed.isActive = !breed.isActive;
+      return breed;
+    },
+    ifBreedActive(breed, id) {
       if (breed.isActive) {
         this.indexClicked++;
         this.lastBreedState.push(id);
+        console.log(this.lastBreedState);
+        return breed;
       }
+      return breed;
+    },
+    setBreedIsActive(breed) {
       this.displayedList.forEach((item) => {
         if (item.name !== breed.name) {
           item.isActive = false;
+          return item;
         }
+        return item;

```

Comment:
Tako nekako sem se zadeve lotil

Reply:
kompozicijo je težje delat če imaš fn ki sprejme več argumentov.. probej še to razbit da ima samo en argument (za vajo, drgač je uredu)